calypso/service.go

```diff
@@ -711,36 +746,49 @@ func (s *Service) NewProtocol(tn *onet.TreeNodeInstance, conf *onet.GenericConfi
 
 		// Wait for DKG in reshare mode to end
 		go func(id byzcoin.InstanceID) {
+			// TODO: properly propagate errors during execution of DKG protocol
+			// (see dedis/cothority#2320)
 			<-setupDKG.Finished
+
+			setValidPeers := func(r *onet.Roster) {
```

Comment:
I think this should be called in _all_ error-returns under `case calypsoReshareProto`, so also on line 743, 715,707, 704. Does that make sense?

Reply:
Indeed, that makes sense.
In the same spirit, shouldn't it also be called in all the error cases in `ReshareLTS()`, such as lines 406, 409, 418, etc? (I could not find where protocol execution in general is described, a pointer to the documentation is welcome :) )

Reply:
I think the protocol itself is not described in the code. You'd have to deep-dive into the calypso-paper and follow the references to the rabin/petersen (I'm not even sure which one is used) to understand what's really going on.

Also, we don't have a real use-case for the moment. So any of the following is OK for me:
1. stop here - the happy-path works, and we don't have a real use-case for it for the moment
2. just add the `SetValidPeers` to `NewProtocol` in case of error
3. do the same for `ReshareLTS`

Reply:
> I think this should be called in _all_ error-returns under `case calypsoReshareProto`, so also on line 743, 715,707, 704. Does that make sense?

Actually, for 704 and 707, I don't yet have the ltsID, so I can't access the previous Roster

Reply:
Actually, I think I will go with 1), because the more I think about it, the more things should be changed (error in 743 is in a function executed later in `pedersen`, an error in 339 means we have to rollback the changes on the successful nodes, etc.)